File: Unidad 1/Ejercicios PROLOG/practica02.pl
/* Práctica 2. Estructura y componentes de los programas para IA en el lenguaje seleccionado.
José Misael Adame Sandoval 
En este Programa definiremos las funciones (relaciones) para dar solución a los reactivos de la Práctica 2.
*/

/*1.- Elabore un programa para determinar el mayor de dos números o indicar que son iguales si es el
caso. Teclee y pruebe los programas en lso respectivos entornos.
Nota: Prolog y Python tienen una funcion max() incorporada, pero hacerla nosotros mismos es buen 
ejercicio ilustrativo. */

max2(X,Y,Z) :- (X =< Y -> Z = Y ; Z = X).

/* 2.- Definir una función max_de_tres(), que tome tres números como argumentos y devuelva el mayor de
ellos. */

max3(N1,N2,N3,Mayor) :- max2(N1,N2,M1), max2(M1, N3, Mayor).

File: Unidad 1/Ejercicios Python/practica02.py
"""Ejericio 1"""
print("Ejericio 1")

n1 = int(input("Ingresa el número 1: "))
n2 = int(input("Ingresa el número 2: "))

if n1 > n2:
    print(n1, "es mayor que", n2)
elif n2 == n1:
    print("Ambos números son iguales")
else:
    print(n2, "es mayor que", n1)

"""Ejercicio 2"""
print("\n\nEjercicio 2")

def max_de_tres(n1, n2, n3):
    if n1 > (n2 and n3):
        print(n1, "es mayor que", n2, "y", n3)
    elif n2 > (n1 and n3):
        print(n2, "es mayor que", n1, "y", n3)
    elif n3 > (n2 and n1):
        print(n3, "es mayor que", n2, "y", n1)
    elif n1 == n2 == n3:
        print("Los tres números son iguales")

max_de_tres(1, 2, 3)
max_de_tres(2, 1, 0)
max_de_tres(3, 8, 6)
max_de_tres(3, 3, 3)
